src/main/java/me/moodcat/database/BulkInsertData.java

```diff
@@ -0,0 +1,219 @@
+import me.moodcat.soundcloud.SoundCloudTrack;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.google.inject.Singleton;
+import com.google.inject.persist.Transactional;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.sql.ordering.antlr.Factory;
+
+/**
+ * MockData inserts initial data in to a clean database.
+ *
+ * @author Jan-Willem Gmelig Meyling
+ */
+@Singleton
+public class BulkInsertData {
+
+    /**
+     * The default [0,0] VA vector.
+     */
+    private static final VAVector DEFAULT_VECTOR = new VAVector(0, 0);
+
+    /**
+     * The path to a newline seperated list of SoundCloud ids.
+     */
+    private static final String SOUNDCLOUD_ID_FILE_PATH = "./src/main/resources/soundcloud_ids";
+
+    /**
+     * The artist DAO provider.
+     */
+    private final Provider<ArtistDAO> artistDAOProvider;
+
+    /**
+     * The song DAO provider.
+     */
+    private final Provider<SongDAO> songDAOProvider;
+
+    /**
+     * The room DAO provider.
+     */
+    private final Provider<RoomDAO> roomDAOProvider;
+
+    /**
+     * The SoundCloud extraction API.
+     */
+    private final SoundCloudExtract soundCloudExtract;
+
+    /**
+     * Object to bulk insert a list of given songs into the databse.
+     * 
+     * @param artistDAOProvider
+     *            the artist provider.
+     * @param songDAOProvider
+     *            the song provider.
+     * @param roomDAOProvider
+     *            the room provider.
+     */
+    @Inject
+    public BulkInsertData(final Provider<ArtistDAO> artistDAOProvider,
+            final Provider<SongDAO> songDAOProvider,
+            final Provider<RoomDAO> roomDAOProvider) {
+        this.artistDAOProvider = artistDAOProvider;
+        this.songDAOProvider = songDAOProvider;
+        this.roomDAOProvider = roomDAOProvider;
+        soundCloudExtract = new SoundCloudExtract();
+    }
+
+    /**
+     * Retrieve songs from the API and put the in the database.
+     * 
+     * @throws IOException
+     *             when the file with song ids could not be parsed.
+     * @throws SoundCloudException
+     *             when the SoundCloud API was not called successfully.
+     */
+    @Transactional
+    public void insertData() throws IOException, SoundCloudException {
+        final SongDAO songDAO = songDAOProvider.get();
+        HashMap<String, Artist> artistMap = new HashMap<>();
+
+        int[] soundCloudIds = readSoundCloudIds();
+        for (int id : soundCloudIds) {
+            try {
+                SoundCloudTrack track = soundCloudExtract.extract(id);
+                Artist artist = getOrPersistArtist(track.getUser().getUsername(), artistMap);
+                Song song = songToTrack(track, id, artist, DEFAULT_VECTOR);
+                songDAO.persist(song);
+            } catch (Exception e) {
+                System.out.println(e.getLocalizedMessage());
```

Comment:
The log statement could be more clear. Maybe we should stick to the logger interface, if we ever want to log to files or something. The `@Slf4j` annotation can be added to the class, from then you can use the static `log` variable:
```
log.info("The song {} could not be persisted due to an error.", e, id);
```